src/Autocomplete/src/Form/ParentEntityAutocompleteType.php

```diff
@@ -35,13 +35,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $formType = $builder->getType()->getInnerType();
         $attribute = AsEntityAutocompleteField::getInstance(\get_class($formType));
-        if (!$attribute) {
-            throw new \LogicException(sprintf('The %s class must have a #[AsEntityAutocompleteField] attribute above its class.', \get_class($formType)));
+
+        if (!$attribute && !isset($options['autocomplete_url'])) {
+            throw new \LogicException(sprintf('You must either provide your own autocomplete_url, or add #[AsEntityAutocompleteField] attribute to %s.', \get_class($formType)));
         }
 
-        $autocompleteUrl = $this->urlGenerator->generate('ux_entity_autocomplete', [
+        // Use the provided URL, or auto-generate from the provided alias
+        $autocompleteUrl = $options['autocomplete_url'] ?? $this->urlGenerator->generate('ux_entity_autocomplete', [
```

Comment:
Perhaps use `isset` here so it's consistent with the check on line 39?

I don't see there's any static analysis here, so a falsey value like empty string would fail the check (empty string *could* be used to call back to same url... unlikely, but potentially...).

Reply:
I'd say there are many ways to to this that are not wrong. I hear you, though. I swapped the 39 line check from `isset` to `empty` to cover all falsy scenarios. Why not.